pkg/quarterdeck/mock/quarterdeck.go

```diff
@@ -107,21 +108,19 @@ func handler(opts ...HandlerOption) http.HandlerFunc {
 		return conf.handler
 	}
 
-	// Encode the fixture data
-	var data []byte
-	if conf.fixture != nil {
-		var err error
-		if data, err = json.Marshal(conf.fixture); err != nil {
-			panic(err)
-		}
-	}
-
 	return func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(conf.status)
-
-		if data != nil {
-			w.Header().Set("Content-Type", "application/json")
-			w.Write(data)
+		switch {
+		case conf.auth && r.Header.Get("Authorization") == "":
+			// TODO: Validate the auth token
+			w.Header().Set("Content-Type", "text/plain")
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte("missing authorization header"))
```

Comment:
Updated the quarterdeck mock so we can test verify that the client is actually sending the authorization token to quarterdeck in tests.